python/ray/client_builder.py

```diff
@@ -100,19 +107,36 @@ def connect(self) -> ClientContext:
                 includes the server's version of Python & Ray as well as the
                 dashboard_url.
         """
+        old_ray_cxt = None
+        if ray.util.client.ray.is_connected() and self._allow_multiple_cli:
```

Comment:
```suggestion
        if ray.util.client.ray.is_connected() and not self._allow_multiple_cli:
```

Reply:
In this places, it's not entering the multiple cli cases.

We have two type of contexts here: without with statement, it's always the global one.

What it's checking here is that if we previously connected with a single client mode, we are not allowed to connect to multiple cli. So here, it's not `not`

If it's multiple client mode, ray.util.client.ray.is_connect() will always be false. Only cli.is_connect() will return true.